apps/network-alerts/test/unit/index_test.js

```diff
@@ -33,54 +41,99 @@ suite('Network Alerts - cellbroadcast system message handling', function() {
   });
 
   test('do not open attention screen if error returns from settings DB',
-    function() {
+    function(done) {
 
     var message = {
       serviceId: 0,
-      messageId: 4370,
+      messageId: 4371,
       body: 'Some body'
     };
 
-    handlerStub.withArgs('cellbroadcast-received').yield(message);
-    MockNavigatorSettings.mRequests[0].onerror();
+    getStub.returns(Promise.reject(new Error('error')));
 
-    sinon.assert.notCalled(window.open);
-    sinon.assert.called(window.close);
-  });
-
-  test('do not open attention screen if CMAS in settings DB is off',
-    function() {
-
-    var message = {
-      serviceId: 0,
-      messageId: 4370,
-      body: 'Some body'
-    };
+    var handler = handlerStub.withArgs('cellbroadcast-received').args[0][1];
+    var finished = handler(message);
 
-    handlerStub.withArgs('cellbroadcast-received').yield(message);
-    MockNavigatorSettings.mRequests[0].result[CMAS_KEY] = {
-      0: false,
-      1: false
-    };
-    MockNavigatorSettings.mReplyToRequests();
+    finished.then(() => {
+      sinon.assert.notCalled(window.open);
+      sinon.assert.called(window.close);
+    }).then(done, done);
+  });
 
-    sinon.assert.notCalled(window.open);
-    sinon.assert.called(window.close);
+  suite('if CMAS in settings DB is off', function() {
+    setup(function() {
+      getStub.returns(Promise.resolve({
+        [CMAS_KEY]: {
+          0: false,
+          1: false
+        }
+      }));
+    });
+
+    test('do not open attention screen', function(done) {
+
+      var message = {
+        serviceId: 0,
+        messageId: 4371,
+        body: 'Some body'
+      };
+
+      var handler = handlerStub.withArgs('cellbroadcast-received').args[0][1];
+      var finished = handler(message);
+
+      finished.then(() => {
+        sinon.assert.notCalled(window.open);
+        sinon.assert.called(window.close);
+      }).then(done, done);
+    });
+
+    [4370, 4383].forEach((presidentialId) => {
+      test('opens an attention screen if message is a presidential alert ' +
+        presidentialId,
+        function(done) {
+
+        var message = {
+          serviceId: 0,
+          messageId: presidentialId,
+          body: 'Some body'
+        };
+
+        var handler = handlerStub.withArgs('cellbroadcast-received').args[0][1];
+        var finished = handler(message);
+
+        var expectedUrl = [
+          'attention.html?',
+          'title=emergency-alert-title&',
+          'body=Some%20body'
+        ].join('');
+
+        finished.then(() => {
+          sinon.assert.calledWith(
+            window.open,
+            expectedUrl, '_blank', 'attention'
+          );
+          sinon.assert.notCalled(window.close);
+        }).then(done, done);
```

Comment:
In my first try I was calling "isPresentialAlert" after the settings check, and that's why the test is asynchronous here. So I kept it asynchronous because I thought it was a more robust check anyway.